CommonFindPackage.cmake

```diff
@@ -269,4 +269,8 @@ macro(common_find_package_post)
   endif()
   common_graph(${PROJECT_NAME})
   message(STATUS ${__configure_msg})
+
+  # If this variable was given in the command line, ensure that the package
+  # installation is only run in this cmake invocation.
+  unset(INSTALL_PACKAGES CACHE)
```

Comment:
try adding the following at the end of SubProject.cmake and see the output of the configure step:

if(NOT DEFINED ${PROJECT_NAME}_IS_SUBPROJECT)
    message(WARNING "SubProject.cmake done: ${PROJECT_NAME} - subproject: ${${PROJECT_NAME}_IS_SUBPROJECT}")
endif()